Sources/SwifterSwift/UIKit/UIColorExtensions.swift

```diff
@@ -3,6 +3,30 @@
 #if canImport(UIKit)
 import UIKit
 
+/// SwifterSwift: Is an enumeration that represents potential errors related to converting hexadecimal strings to colors.
+///
+/// This enumeration is used to capture and report errors that may occur when attempting to initialize a color using a hexadecimal string.
+/// It handles cases where the provided string either does not have the correct length or contains invalid characters.
+///
+/// - Case `invalidHexLength`: Indicates that the provided hexadecimal string does not have exactly 6 characters,
+///   which is required for a valid conversion.
+/// - Case `invalidHexFormat`: Indicates that the provided hexadecimal string contains characters that are not valid
+///   hexadecimal digits (0-9, A-F).
+enum ColorError: Error {
```

Comment:
I know it is against the norm, but here on this project we try to keep extensions self contained, so we don't introduce new types.